More updates to testing documentation (#45528)
dotnet/docs: docs/core/testing/microsoft-testing-platform-integration-dotnet-test.md

---
title: Use Microsoft.Testing.Platform with VSTest mode of `dotnet test`
description: Learn how to run Microsoft.Testing.Platform tests through dotnet test.
author: nohwnd
ms.author: jajares
ms.date: 03/26/2025
---

# Use Microsoft.Testing.Platform with VSTest mode of `dotnet test`

This article describes `dotnet test` integration for Microsoft.Testing.Platform that is provided by Microsoft.Testing.Platform.MSBuild when running with VSTest mode of `dotnet test`.

> Before reading this article, it's advised to read [Testing with dotnet test](unit-testing-with-dotnet-test.md) first.

Reviewer: ```suggestion
Before diving into this article, it's recommended to first read [Testing with dotnet test](unit-testing-with-dotnet-test.md) for better understanding.
```

> By default, `dotnet test` is using VSTest behavior to run tests. To enable support for `Microsoft.Testing.Platform` in `dotnet test`, there are two options:

Reviewer: ```suggestion
By default, `dotnet test` uses VSTest to run tests. To enable support for Microsoft.Testing.Platform in `dotnet test`, you have two options:
```

> 1. Use `dotnet test` in VSTest mode, and specify `<TestingPlatformDotnetTestSupport>true</TestingPlatformDotnetTestSupport>` MSBuild property in your project file.

Reviewer: ```suggestion
1. Use `dotnet test` in VSTest mode and specify the `<TestingPlatformDotnetTestSupport>true</TestingPlatformDotnetTestSupport>` MSBuild property in your project file.
```

> 2. Use `dotnet test` in MTP mode, and enjoy more native support of MTP in `dotnet test`. This is only supported starting with .NET 10 SDK.

Reviewer: ```suggestion
2. Use `dotnet test` in MTP mode for more native support of MTP in `dotnet test`, which is only supported starting with the .NET 10 SDK.
```

> Both of these options are explained in details in the [Testing with dotnet test](unit-testing-with-dotnet-test.md) article.

Reviewer: ```suggestion
Both options are explained in detail in the [Testing with dotnet test](unit-testing-with-dotnet-test.md) article.
```

> [!IMPORTANT]

> > The remaining of this article is specific to the first option.

Reviewer: ```suggestion
> The rest of this article focuses specifically on the first option.
```

Author: I'm not sure, but I feel "focus" may give impression that small parts of the article may apply to the second option.

Reviewer: It's up to you if you'd like to change it :)
